internal/liqonet/tunnel-operator/reflected_endpointslice_controller.go

```diff
@@ -0,0 +1,283 @@
+	// Create the IPTables handler
+	iptablesHandler, err := liqoiptables.NewIPTHandler()
+	if err != nil {
+		return nil, err
+	}
+	// Create the IPSet handler
+	ipsetHandler := liqoipset.NewIPSetHandler()
+	// Create and return the controller
+	return &reflectedEndpointsliceController{
+		Client:             cl,
+		Scheme:             scheme,
+		IPTHandler:         iptablesHandler,
+		IPSetHandler:       &ipsetHandler,
+		gatewayNetns:       gatewayNetns,
+		podsInfo:           podsInfo,
+		endpointslicesInfo: endpointslicesInfo,
+	}, nil
+}
+
+// Reconcile local endpointslices that are also reflected on remote clusters as a result of offloading
+func (r *reflectedEndpointsliceController) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	_ = log.FromContext(ctx)
+
+	nsName := req.NamespacedName
+	klog.Infof("Reconcile Endpointslice %q", nsName)
+
+	endpointslice := discoveryv1.EndpointSlice{}
+	if err := r.Get(ctx, nsName, &endpointslice); err != nil {
+		err = client.IgnoreNotFound(err)
+		if err == nil {
+			// Endpointslice not found, endpointsliceInfo object found: delete endpointInfo objects
+			if value, ok := r.endpointslicesInfo.LoadAndDelete(nsName); ok {
+				klog.Infof("Endpointslice %q not found: ensuring updated iptables rules", nsName)
+
+				// Soft delete object
+				endpointsInfo := value.(map[string]liqoiptables.EndpointInfo)
+				for endpoint, endpointInfo := range endpointsInfo {
+					endpointInfo.Deleting = true
+					endpointsInfo[endpoint] = endpointInfo
+				}
+				r.endpointslicesInfo.Store(nsName, endpointsInfo)
+
+				if err := r.gatewayNetns.Do(r.ensureIptablesRules); err != nil {
+					klog.Errorf("Error while ensuring iptables rules: %w", err)
+					return ctrl.Result{}, err
+				}
+
+				// Hard delete object
+				r.endpointslicesInfo.Delete(nsName)
+			}
+		}
+		return ctrl.Result{}, err
+	}
+
+	// Check endpointslice's namespace offloading
+	nsOffloading, err := getters.GetOffloadingByNamespace(ctx, r.Client, endpointslice.Namespace)
+	if err != nil {
+		klog.Error(err)
+		err = client.IgnoreNotFound(err)
+		if err == nil {
+			// Delete endpointInfo objects related to this endpointslice
+			if value, ok := r.endpointslicesInfo.LoadAndDelete(nsName); ok {
+				// Endpointslice not found, endpointsliceInfo object found: ensure iptables rules
+				klog.Infof("Endpointslice %q not found: ensuring updated iptables rules", nsName)
+
+				// Soft delete object
+				endpointsInfo := value.(map[string]liqoiptables.EndpointInfo)
+				for endpoint, endpointInfo := range endpointsInfo {
+					endpointInfo.Deleting = true
+					endpointsInfo[endpoint] = endpointInfo
+				}
+				r.endpointslicesInfo.Store(nsName, endpointsInfo)
+
+				if err := r.gatewayNetns.Do(r.ensureIptablesRules); err != nil {
+					klog.Errorf("Error while ensuring iptables rules: %w", err)
```

Comment:
You don't need to log it, just create a new error with fmt.Errorf and return it.